### PR TITLE
SJQ2T-372 BE - Create Tag Endpoint

### PR DESCRIPTION
**Pull request by MarkoRadunovicHtec**
Endpoint for create Tag added, model classes added, flyway script added, Need to add unit tests.

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/controller/TagController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@RestController
+@RequestMapping("/tags")
+@AllArgsConstructor
+@Tag(name = "Tag API", description = "Tag-related operations")
+public class TagController {
+
+  private TagService tagService;
+
+  @PostMapping
+  public ResponseEntity<TagResponse> createTag(
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/controller/TagController.java:107]: don't forget to add docs later

File: backend/src/main/java/com/htecgroup/skynest/model/entity/TagEntity.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class TagEntity implements Serializable {
+
+  @Id @GeneratedValue private UUID id;
+
+  @ManyToOne
+  @JoinColumn(name = "company_id", nullable = false)
+  private CompanyEntity company;
+
+  String name;
+  String rgb;
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/model/entity/TagEntity.java]: shouldn't these be private?

File: backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    tagEntity.setName(tagEntity.getName().trim());
+    tagEntity.setCompany(currentUsersCompany);
+    tagEntity.setRgb(tagEntity.getRgb().substring(1));
+
+    TagEntity savedTagEntity = tagRepository.save(tagEntity);
+
+    log.info(
+        "User {} ({}) has created tag {} ({})",
+        currentUserService.getLoggedUser().getName(),
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java]: probably should be email

File: backend/src/main/resources/db/migration/V14.1__tag_create.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `tag` (
+  `id` binary(16) NOT NULL DEFAULT (uuid_to_bin(uuid())),
+  `company_id` binary(16) NOT NULL DEFAULT (uuid_to_bin(uuid())),
```
Comment (ivanfhtec) [inline backend/src/main/resources/db/migration/V14.1__tag_create.sql]: this should not have a DEFAULT

File: backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    TagEntity tagEntity = modelMapper.map(createTagRequest, TagEntity.class);
+
+    CompanyEntity currentUsersCompany =
+        currentUserService
+            .getCompanyEntityFromLoggedUser()
+            .orElseThrow(CompanyNotFoundException::new);
+
+    tagEntity.setName(tagEntity.getName().trim());
+    tagEntity.setCompany(currentUsersCompany);
+    tagEntity.setRgb(tagEntity.getRgb().substring(1));
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java]: so we should probably reattach the `#` when returning tags to the FE, whenever that gets implemented

or we could require colours to be `[0-9a-fA-F]{6}`

File: backend/src/main/java/com/htecgroup/skynest/controller/TagController.java (hunk 1 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/controller/TagController.java:107]: Working on it :)

File: backend/src/main/java/com/htecgroup/skynest/model/entity/TagEntity.java (hunk 2 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/model/entity/TagEntity.java]: They should. Dont know how I forgot that.

File: backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java (hunk 3 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java]: Makes more sense

File: backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java (hunk 5 above)
Comment (MarkoRadunovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java]: Regex will be changed to accept only without #

File: backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        currentUserService
+            .getCompanyEntityFromLoggedUser()
+            .orElseThrow(CompanyNotFoundException::new);
+
+    tagEntity.setName(tagEntity.getName().trim());
+    tagEntity.setCompany(currentUsersCompany);
+
+    TagEntity savedTagEntity = tagRepository.save(tagEntity);
+
+    log.info(
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java:47]: I like the in depth logs

**Review by markobarackovhtec — APPROVED** [review]

**Pull request merged**